apps/tangle-dapp/data/blueprints/useBlueprintListing.ts

```diff
@@ -0,0 +1,75 @@
+import type { Option } from '@polkadot/types';
+import type { TanglePrimitivesServicesOperatorPreferences } from '@polkadot/types/lookup';
+import { useCallback } from 'react';
+import { combineLatest, switchMap } from 'rxjs';
+
+import useNetworkStore from '../../context/useNetworkStore';
+import useApiRx from '../../hooks/useApiRx';
+import { TangleError, TangleErrorCode } from '../../types/error';
+import { useOperatorTVL } from '../restake/useOperatorTVL';
+import useRestakeAssetMap from '../restake/useRestakeAssetMap';
+import useRestakeOperatorMap from '../restake/useRestakeOperatorMap';
+import {
+  createBlueprintObjects,
+  extractBlueprintsData,
+  extractOperatorData,
+  fetchOwnerIdentities,
+} from './utils/blueprintHelpers';
+
+export default function useBlueprintListing() {
+  const { rpcEndpoint } = useNetworkStore();
+  const { operatorMap } = useRestakeOperatorMap();
+  const { assetMap } = useRestakeAssetMap();
+  const { operatorTVL } = useOperatorTVL(operatorMap, assetMap);
+
+  const { result, ...rest } = useApiRx(
+    useCallback(
+      (apiRx) => {
+        if (
+          apiRx.query.services?.blueprints === undefined ||
+          apiRx.query.services?.operators === undefined
+        )
+          // TODO: Should return the error here instead of throw it
```

Comment:
@drewstone I'm planning to modify the `useApiRx` hook to be able to manage errors by returning them from functions like these briefly, when I do that I'll handle this TODO

Reply:
@AtelyPham In this specific case, since it's checking whether the pallets are available, if the feature isn't supported we should probably have an error state or at least default to 'empty,' since this is a possible edge case that the user would run into. For example, simply switching to mainnet (where the pallet(s) might not be live yet) would trigger this error to be thrown

Reply:
When this error occurs, the UI handles it by displaying a message to inform users that the feature is not supported.